### PR TITLE
Skills docs

### LINKED ISSUE #105: Skill Readme

### PR DESCRIPTION
**Pull request by lizgzil**
---

# Description

Update to the skills docs.

Fixes #105 

# Instructions for Reviewer

In order to test the code in this PR you need to ...

Please pay special attention to ...

# Checklist:

- [ ] I have refactored my code out from `notebooks/`
- [ ] I have checked the code runs
- [ ] I have tested the code
- [ ] I have run `pre-commit` and addressed any issues not automatically fixed
- [ ] I have merged any new changes from `dev`
- [ ] I have documented the code
  - [ ] Major functions have docstrings
  - [ ] Appropriate information has been added to `README`s
- [ ] I have explained this PR above
- [ ] I have requested a code review

### CODE REVIEW

File: dap_prinz_green_jobs/pipeline/green_measures/skills/README.md [hunk 1]
```diff
@@ -1,62 +1,115 @@
-# Green Skills
+# 🧑‍🔧🥬 Green Skills
```
Comment (india-kerle) [inline dap_prinz_green_jobs/pipeline/green_measures/skills/README.md:1]: the emoji choice is *chefs kiss*

File: dap_prinz_green_jobs/pipeline/green_measures/skills/README.md [hunk 2]
```diff
@@ -1,62 +1,115 @@
...
 
-This directory contains the scripts needed to extract green skills for a given job advert.
+This directory contains the scripts needed to extract green skills for a given job advert. This directory is organised as follows:
 
-The script
+1. [skills_measures_utils.py](https://github.com/nestauk/dap_prinz_green_jobs/tree/dev/dap_prinz_green_jobs/pipeline/green_measures/skills/skills_measures_utils.py): The script which contains the `SkillMeasures` class to find green skills in job adverts. This utilises our [Skills Extractor library](https://github.com/nestauk/ojd_daps_skills) to extract and map skills onto a structured green skills list.
```
Comment (india-kerle) [inline dap_prinz_green_jobs/pipeline/green_measures/skills/README.md:5]: nice shout out to ojd daps skills

File: dap_prinz_green_jobs/pipeline/green_measures/skills/README.md [hunk 3]
```diff
@@ -1,62 +1,115 @@
...
+2. [map_skills_utils.py](https://github.com/nestauk/dap_prinz_green_jobs/tree/dev/dap_prinz_green_jobs/pipeline/green_measures/skills/map_skills_utils.py): Utils to match skill embeddings to all of the ESCO taxonomy. The majority of this code is from the [ojd_daps_skills](https://github.com/nestauk/ojd_daps_skills) repo, but replicated here since currently there was no functionality in `ojd_daps_skills` to take in already embedded skills and map them.
+3. [customise_skills_extractor.py](https://github.com/nestauk/dap_prinz_green_jobs/tree/dev/dap_prinz_green_jobs/pipeline/green_measures/skills/customise_skills_extractor.py): To add the custom config file, formatted skills taxonomy and taxonomy embeddings to the relevant location in the `ojd-daps-skills` library location run this script with the `--config_name "extract_green_skills_esco"` argument. NOTE: it isn't neccessary to run this anymore, as such this file is unneeded.
```
Comment (india-kerle) [inline dap_prinz_green_jobs/pipeline/green_measures/skills/README.md]: hmmm should we just delete this file then?

File: dap_prinz_green_jobs/pipeline/green_measures/skills/README.md [hunk 4]
```diff
@@ -1,62 +1,115 @@
...
-weighted avg       0.91      0.91      0.91       429
+**Step 1:** Skills are extracted from the job advert by applying the trained skills extractor model. Skill entities over 10 words long are split up into separate entities.
+
+**Step 2:** The skills extracted, all the green ESCO skills and the all the O\*NET green topics are embedded using the `all-MiniLM-L6-v2` Sentence Tranformers pretrained model. The cosine similarity scores are then calculated, allowing us to find the closest green ESCO skill, the closest O\*NET green topic, and the number of O\*NET green topics exactly found in the entity (exact word matching).
+
+**Step 3:** A trained green skill classifier model is then applied using the features found from the previous step. This predicts whether the skill entity is green or not. If the skill is green, then the details of the closest green ESCO skill are outputted.
+
+### 🏋️ Training a Green Skills Classifier
 
+Core to `SkillMeasures` is the predicting whether a skill entity is likely to be green or not, this is done by training a random forest classifier. The green skills classifier can be trained by running:
```
Comment (india-kerle) [inline dap_prinz_green_jobs/pipeline/green_measures/skills/README.md]:
```suggestion
Core to `SkillMeasures` is predicting whether a skill entity is likely to be green or not. This is done by training a random forest classifier. The green skills classifier can be trained by running:
```

File: dap_prinz_green_jobs/pipeline/green_measures/skills/README.md [hunk 5]
```diff
@@ -1,62 +1,115 @@
...
+
+```
+python dap_prinz_green_jobs/pipeline/green_measures/skills/green_skill_classifier.py
 ```
 
+The training data for this is `s3://prinz-green-jobs/inputs/data/training_data/green_skill_training_data.csv` - this was created by labelling a dataset of frequently occurring skills, as well as skills which are mapped to ESCO green skills with a high similarity score.
```
Comment (india-kerle) [inline dap_prinz_green_jobs/pipeline/green_measures/skills/README.md]:
```suggestion
The training data for this is found in `s3://prinz-green-jobs/inputs/data/training_data/green_skill_training_data.csv` - this was created by labelling a dataset of frequently occurring skills, as well as skills which are mapped to ESCO green skills with a high similarity score.
```

File: dap_prinz_green_jobs/pipeline/green_measures/skills/skill_measures_utils.py [hunk 6]
```diff
@@ -403,6 +403,12 @@ def get_measures(
             dict: A dictionary of job advert ids and green measures information
         """
 
+        num_unique_ids = len(set([s[job_id_key] for s in job_adverts]))
+        if len(job_adverts) != num_unique_ids:
+            logger.warning(
+                f"The job advert IDs are not always unique - duplicated IDs will be overwritten - consider re-inputting job_adverts with unique IDs"
+            )
```
Comment (india-kerle) [inline dap_prinz_green_jobs/pipeline/green_measures/skills/skill_measures_utils.py:410]: nice - this is super helpful

**Review by india-kerle — APPROVED** [review]
looks great!

File: dap_prinz_green_jobs/pipeline/green_measures/skills/README.md (hunk 3 above)
Comment (lizgzil) [reply dap_prinz_green_jobs/pipeline/green_measures/skills/README.md]: yeh I guess so .. I suppose its a bit confusing to keep it

**Comment by lizgzil** [comment]
thanks @india-kerle ! changes made

**Pull request merged**